[2023-11-21 15:10:24,664] {taskinstance.py:1179} INFO - Dependencies all met for <TaskInstance: etl_hh.load_core_dim_currency scheduled__2023-11-20T00:00:00+00:00 [queued]>
[2023-11-21 15:10:24,710] {taskinstance.py:1179} INFO - Dependencies all met for <TaskInstance: etl_hh.load_core_dim_currency scheduled__2023-11-20T00:00:00+00:00 [queued]>
[2023-11-21 15:10:24,713] {taskinstance.py:1376} INFO - 
--------------------------------------------------------------------------------
[2023-11-21 15:10:24,725] {taskinstance.py:1377} INFO - Starting attempt 2 of 2
[2023-11-21 15:10:24,726] {taskinstance.py:1378} INFO - 
--------------------------------------------------------------------------------
[2023-11-21 15:10:24,767] {taskinstance.py:1397} INFO - Executing <Task(PostgresOperator): load_core_dim_currency> on 2023-11-20 00:00:00+00:00
[2023-11-21 15:10:24,776] {standard_task_runner.py:52} INFO - Started process 1239 to run task
[2023-11-21 15:10:24,782] {standard_task_runner.py:79} INFO - Running: ['***', 'tasks', 'run', 'etl_hh', 'load_core_dim_currency', 'scheduled__2023-11-20T00:00:00+00:00', '--job-id', '261', '--raw', '--subdir', 'DAGS_FOLDER/pipeline.py', '--cfg-path', '/tmp/tmprs7dpyh1', '--error-file', '/tmp/tmpo09lw2od']
[2023-11-21 15:10:24,783] {standard_task_runner.py:80} INFO - Job 261: Subtask load_core_dim_currency
[2023-11-21 15:10:24,951] {task_command.py:371} INFO - Running <TaskInstance: etl_hh.load_core_dim_currency scheduled__2023-11-20T00:00:00+00:00 [running]> on host f44f09cebae7
[2023-11-21 15:10:25,157] {taskinstance.py:1591} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=etl_hh
AIRFLOW_CTX_TASK_ID=load_core_dim_currency
AIRFLOW_CTX_EXECUTION_DATE=2023-11-20T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=2
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2023-11-20T00:00:00+00:00
[2023-11-21 15:10:25,190] {base.py:68} INFO - Using connection ID 'my_db_conn' for task execution.
[2023-11-21 15:10:25,260] {dbapi.py:231} INFO - Running statement: 
        MERGE INTO core.dim_currency AS trg
        USING(
            SELECT DISTINCT 
                salary_currency,
                CASE
                    WHEN salary_currency = 'RUR'
                    THEN 1
                    WHEN salary_currency = 'USD'
                    THEN 90
                    WHEN salary_currency = 'KZT'
                    THEN 0.2
                    WHEN salary_currency = 'BYR'
                    THEN 28.7
                END AS to_rub
            FROM staging.vacancy) AS src
        ON trg.currency_id = src.salary_currency
        WHEN NOT MATCHED
        THEN INSERT(
            currency_id, 
            to_rub
            )
        VALUES(
            src.salary_currency,
            src.to_rub
            )
        WHEN MATCHED AND COALESCE(trg.to_rub, '') <> COALESCE(src.to_rub, '')
        THEN UPDATE
        SET 
            currency_id = src.salary_currency, 
            to_rub = src.to_rub, parameters: None
[2023-11-21 15:10:25,359] {taskinstance.py:1909} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/postgres/operators/postgres.py", line 92, in execute
    self.hook.run(self.sql, self.autocommit, parameters=self.parameters)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/hooks/dbapi.py", line 211, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/hooks/dbapi.py", line 235, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.InvalidTextRepresentation: invalid input syntax for type numeric: ""
LINE 27:         WHEN MATCHED AND COALESCE(trg.to_rub, '') <> COALESC...
                                                       ^

[2023-11-21 15:10:25,412] {taskinstance.py:1420} INFO - Marking task as FAILED. dag_id=etl_hh, task_id=load_core_dim_currency, execution_date=20231120T000000, start_date=20231121T151024, end_date=20231121T151025
[2023-11-21 15:10:25,481] {standard_task_runner.py:97} ERROR - Failed to execute job 261 for task load_core_dim_currency (invalid input syntax for type numeric: ""
LINE 27:         WHEN MATCHED AND COALESCE(trg.to_rub, '') <> COALESC...
                                                       ^
; 1239)
[2023-11-21 15:10:25,518] {local_task_job.py:156} INFO - Task exited with return code 1
[2023-11-21 15:10:25,719] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
